File: app/src/main/java/soup/movie/ui/theater/sort/TheaterSortContract.java
package soup.movie.ui.theater.sort;

import android.support.annotation.NonNull;

import java.util.List;

import soup.movie.data.model.Theater;
import soup.movie.ui.BaseContract;

public interface TheaterSortContract {

    interface Presenter extends BaseContract.Presenter<View> {
        void onConfirmClicked(List<Theater> selectedTheaters);
    }

    interface View extends BaseContract.View {
        void render(@NonNull TheaterSortViewState viewState);
    }
}

File: app/src/main/java/soup/movie/ui/theater/sort/TheaterSortContract.kt
package soup.movie.ui.theater.sort

import soup.movie.ui.LegacyBaseContract

interface TheaterSortContract {

    interface Presenter : LegacyBaseContract.Presenter<View> {

        fun onItemMove(fromPosition: Int, toPosition: Int)

        fun saveSnapshot()
    }

    interface View : LegacyBaseContract.View {

        fun render(viewState: TheaterSortViewState)
    }
}
